graph/patterns/change-status.md

```diff
@@ -0,0 +1,86 @@
+# Change Status & Action Request
+
+Microsoft Graph API Design Pattern
+
+_The change status pattern is an effective alternative to model the ability to trigger side effects while keeping an agnostic, data driven approach_
+
+## Problem
+
+In some cases, exposed resources can trigger side effects that modify the system behavior. For instance, a [`riskyUser`](https://docs.microsoft.com/en-us/graph/api/resources/riskyuser?view=graph-rest-1.0) can be [`confirmed`](https://docs.microsoft.com/en-us/graph/api/riskyuser-confirmcompromised?view=graph-rest-1.0&tabs=http) or [`dismissed`](https://docs.microsoft.com/en-us/graph/api/riskyuser-confirmcompromised?view=graph-rest-1.0&tabs=http).
+
+These situation are often modeled as a `POST` request to a specific endpoint in the same URI space of the resource, triggering the side effect directly and not returning any data. While it might be a quick and easy solution, there are some considerations:
+
+- The API should be exposing resources and allow their manipulation, while Actions resemble an object in [OOP][1]
+- It couples the request with the side effect, making more difficult to test, log, replay and reason about what is happening in the system
+- It forces the consumer to follow the [OOP][1] approach at the API level. We do not want to pursue any specific style and the developers writing the client might be working in other ways
+
+## Solution
+
+API Designers can prefer the usage of a `status` property or a datafyed version of the intended action to represent the **intent** of triggering a side effect rather than starting the execution directly through the HTTP request.
```

Comment:
I have been wondering about it myself. The initiating user can always query the operation since they have a reference, but other people would not. I have added the proposal in - let me know what you think: d85a901

Reply:
OK - so given that object already has "riskyState" any additional property is effectively a desiredState

Reply:
That is correct. I would be fine to "mandate" to call it `desiredState` if you want, but at the moment I do not see any automated tooling that might use that property to do anything useful, thus leaving the naming to the implementer.